### ISSUE TITLE
Bug: Product config function fetchAndActivate is not working as expected.

### ISSUE THREAD

**Issue by SuhasTS** [CleverTap/clevertap-android-sdk#127]
Function `fetchAndActivate()` in class `CTProductConfigController` is fetching the new values but failing to activate the same in some scenarios.
```
fun setup(context: Context) {
        Log.d("ProductExperience", "setup: ")
        val clevertapApi = CleverTapAPI.getDefaultInstance(context) ?: return
        val productConfigController = clevertapApi.productConfig()
        productConfigController.setDefaults(R.xml.product_experience_defaults)
        clevertapApi.setCTProductConfigListener(object : CTProductConfigListener {
            override fun onActivated() {
                Log.d("ProductExperience", "onActivated: ${productConfigController.getString("Debug")}")
            }

            override fun onFetched() {
                Log.d("ProductExperience", "onFetched: ${productConfigController.getString("Debug")}")
            }

            override fun onInit() {
                Log.d("ProductExperience", "onInit: ${productConfigController.getString("Debug")}")
                productConfigController.activate()
                productConfigController.fetchAndActivate()
            }
        })
    }
```
Logs on running the above function 

```
D/ProductExperience: setup: 
D/ProductExperience: onInit: 
D/ProductExperience: onActivated: Default value in xml
D/ProductExperience: onInit: Default value in xml
D/ProductExperience: onActivated: Default value in xml
D/ProductExperience: onFetched: Default value in XML  //onactivated callback is missing after this
```

**Comment by kumaratulismu** [comment]
@SuhasTS Thanks for the time & effort to raise a fix for this. 

Before going ahead with the PR, Can you please explain the scenario in which it's failing & the solution you have proposed will work to handle the scenario?

**Comment by SuhasTS** [comment]
The sample function that I have shared is the scenario where it fails. 
Basically when you call `fetchAndActivate` and previously called `activate` function is completed after that(`activate` function is asynchronous) the variable `isFetchAndActivating` is reset to `false` as a result, it only fetches the new values but never activates them.
The PR which I have raised resolves this issue.

**Comment by kumaratulismu** [comment]
@SuhasTS As per my understanding , you are calling activate() & fetchAndActivate() method one after another in a sequence. So the isFetchAndActivating = true is getting overridden.

Please confirm.

**Comment by SuhasTS** [comment]
Yes.

**Comment by kumaratulismu** [comment]
@SuhasTS Thanks for confirming. 

We will including the fix in our next planned release.
Will keep you informed.

**Comment by kumaratulismu** [comment]
@SuhasTS we have fixed this issue with latest release(4.0.3).

Closing this.
